### PR TITLE
Update styling for Create User Profile desktop

### LINKED ISSUE #176: Update styling for Create User Profile for desktop
- [x] Update current Create User Profile page styling for desktop.

Refer to the latest design in FIGMA under Desktop App MVP.
https://www.figma.com/file/4cV84H1jyMo0rJEAl3HNnJ/FightPandemics-landing-page?node-id=1377%3A8027

- [x]  Designs need to be pixel perfect

### PR DESCRIPTION
**Pull request by T-Allen**
### Quick Description of the PR

Created a CreateUserProfile page and added styling for desktop and mobile.

### An Overview of the Changes
- New page CreateUserProfile.js
- Added 'create-profile' route to routes.js
- Added new image asset (.svg) from Figma to 'create-profile-images' folder in 'assets' folder.
- Installed 'typeface-dm-sans' package in client to display DM Sans font. 

Resolves #176

### CODE REVIEW

File: client/src/pages/CreateUserProfile.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                <DropdownMenu>
+                    <div id="dropdown-anchor" style={{'position': 'relative'}}>
+                        <TextInput label="Address"/>
+                    </div>
+                </DropdownMenu>
+            </InputGroup>
+            <CheckboxGroup description="I am traveling" />
+            <CheckboxGroup description="Don't show my address" />
+            <InputGroup>
+                <Title>I want to</Title>
```
Comment (anniegiang) [inline client/src/pages/CreateUserProfile.js:189]: Need to update the styling a bit:

current:
Screen Shot 2020-05-07 at 2 36 10 PM

figma:
Screen Shot 2020-05-07 at 2 36 19 PM
Screen Shot 2020-05-07 at 2 36 27 PM

File: client/src/pages/CreateUserProfile.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+            <InputGroup>
+                <TextInput label="E-mail" />
+                <TextInput label="Name" />
+                <DropdownMenu>
+                    <div id="dropdown-anchor" style={{'position': 'relative'}}>
+                        <TextInput label="Address"/>
+                    </div>
+                </DropdownMenu>
+            </InputGroup>
+            <CheckboxGroup description="I am traveling" />
```
Comment (anniegiang) [inline client/src/pages/CreateUserProfile.js:186]: Need to update styling abit to match designs:

current:
Screen Shot 2020-05-07 at 2 36 51 PM

figma:
Screen Shot 2020-05-07 at 2 36 47 PM
Screen Shot 2020-05-07 at 2 42 34 PM

Comment (anniegiang) [inline client/src/pages/CreateUserProfile.js:186]: For the checkboxes, need to update styling to match Figma

current:
Screen Shot 2020-05-07 at 2 35 30 PM

figma:
Screen Shot 2020-05-07 at 2 35 34 PM

File: client/src/pages/CreateUserProfile.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+                label="Medical Help" 
+                description="I have symptoms of COVID-19"/>
+                <CheckboxGroup 
+                label="Other Help" 
+                description="I need assitance getting groceries, medicine, etc." />
+            </InputGroup>
+            <InputGroup>
+                <SubmitButton primary="true">Create Profile</SubmitButton>
+                <CheckboxGroup>
+                    <UnderlineLink>By signing up, I agree to the <a href="#">Terms and Conditions</a>
```
Comment (anniegiang) [inline client/src/pages/CreateUserProfile.js:206]: The order of the UnderlineLinks are switched, should be Privacy and Policy first. 

You can find the `href` values in `routes.js` for both urls to both pages.

File: client/src/pages/CreateUserProfile.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        img {
+            margin-top: 16rem;
+        }
+    }
+    .form-container, .image-container {
+        padding: 1rem;
+    }
+`;
+
+const ProfileFormGroup = styled.form`
```
Comment (anniegiang) [inline client/src/pages/CreateUserProfile.js:50]: The background color of the form section looks different than on Figma, I think the color it should be Selago.

File: client/src/pages/CreateUserProfile.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+import UnderlineDescription from "../components/Input/UnderlineDescription";
+import CustomH1 from "../components/Typography/Title/CustomH1";
+import Title from "../components/Typography/Title";
+import SubmitButton from "../components/Button/SubmitButton";
+import { theme, mq } from "../constants/theme";
+
+import PersonalDataImage from "../assets/create-profile-images/personal_data.svg";
+
+require('typeface-dm-sans');
+require('typeface-work-sans');
```
Comment (anniegiang) [inline client/src/pages/CreateUserProfile.js:17]: For some reason I'm unable to `npm install` this so it's blocking the app from running. So for now in order to see the app I commented this out.

File: client/src/pages/CreateUserProfile.js [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+return <Container>
+    <Flex className="image-container" direction="column">
+        <img src={PersonalDataImage} alt="two people standing next to a list of personal items" />
+    </Flex>
+    <Flex className="form-container" direction="column">
+        <WhiteSpace size="xl"/>
+        <ProfileFormGroup>
+            <CustomH1 color={theme.colors.darkerGray} fontweight="bold">Create your Profile</CustomH1>
+            <InputGroup>
+                <TextInput label="E-mail" />
```
Comment (anniegiang) [inline client/src/pages/CreateUserProfile.js:178]: - I'm not sure if it's a mistake by the design team but only the email field is a gray while the other inputs are royal blue.
- The underline moves surrounding elements when it's active/hovered/focused
- Need the icon that's next to the Address label

figma:
Screen Shot 2020-05-07 at 2 49 49 PM

**Comment by T-Allen** [comment]
> Thanks for working on this! I left some comments ready for your review.
> 
> But before you start, is this PR a duplicate of PR #100 by @CeesWang? I believe that PR was only the mobile version, while this PR is both mobile and desktop. Can you guys clarify with each other?

Kinda, I had mistakenly thought there was no mobile page, so I updated the title. I changed it back to reflect that this is only for desktop. I will probably add the desktop styling to @CeesWang's mobile page instead of creating a new page in my revisions.

File: client/src/pages/CreateUserProfile.js (hunk 6 above)
Comment (T-Allen) [reply client/src/pages/CreateUserProfile.js:178]: Yeah, on the Desktop MVP in Figma, there appears to be a screen where the email field is grayed out, and another screen where it is not while the location drop-down menu is open. I am not sure if that is intentional. 

image

**Comment by anniegiang** [comment]
@T-Allen Awesome, sounds good. Thanks for clarifying.

File: client/src/pages/CreateUserProfile.js (hunk 6 above)
Comment (T-Allen) [reply client/src/pages/CreateUserProfile.js:178]: Actually on this screen, the email field is removed entirely and the email address is displayed in a welcome message as a subheading. I'm assuming this is the intended behavior after a user enters their email address.

File: client/src/pages/CreateUserProfile.js (hunk 5 above)
Comment (robinv85) [reply client/src/pages/CreateUserProfile.js:17]: what error do you get? Might be problem with package-lock, or the npm module itself?

**Comment by anniegiang** [comment]
> lgtm, does this actually add the page too? Would be nice to get this merged asap so we can continue with the whole user onboarding flow and easily create users and login to test all the other features

If we're in a rush we could merge this since the changes are minor, and then continue to make the changes after it's merged.

**Comment by T-Allen** [comment]
> lgtm, does this actually add the page too? Would be nice to get this merged asap so we can continue with the whole user onboarding flow and easily create users and login to test all the other features

@robinv85, unfortunately not. I had only written the styled components and layout for the page with the profile creation form. I also am not sure if this page and the confirmation screens belong in the same file, and I had not added those.

**Comment by robinv85** [comment]
@T-Allen what's the status of this PR?
For reference, probably best if we use the `react-hook-form` library and avoid redux but use `useReducer` hooks as much as possible, see my recently opened PR that refactors the login & signup page 415
For now I've only kept the logged-in user in redux with "isauthenticated" and "emailVerified" fields, we will also have to add the check if a user in our db exists to handle the redirection to this page when needed. I would like to help out with this...

**Comment by T-Allen** [comment]
@robinv85 I've addressed each of the style changes except for the the exact shape of the check icon in the check-boxes which does not seem to be a priority right now. Also, I had to target some of the parent elements to hide the toolbar on this screen, as was shown in the designs.  
image

**Comment by robinv85** [comment]
@T-Allen ok cool, can we get this merged asap so we can continue to finish the whole onboarding flow and the rest of the functionalities for logged-in users? because i think we might be missing some more tickets there for things that belong in MVP

**Comment by T-Allen** [comment]
@robinv85 For sure. Unfortunately, this PR is linked to my forked repo which does not have appropriate permissions. I'm going to close this, and resolve the issue with a new PR.

**Pull request closed without merging**